spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2331,19 +2332,17 @@ private void commitOffsetsIfNeeded(final ConsumerRecords<K, V> records) {
 		}
 
 		private void batchInterceptAfter(ConsumerRecords<K, V> records, @Nullable Exception exception) {
-			if (this.commonBatchInterceptor != null) {
-				try {
-					if (exception == null) {
-						this.commonBatchInterceptor.success(records, this.consumer);
-					}
-					else {
-						this.commonBatchInterceptor.failure(records, exception, this.consumer);
-					}
+			try {
```

Comment:
This is incorrect; the interceptor can be null; it probably should be marked `@Nullable`.